WordPress/Classes/ViewRelated/Post/PostSettingsViewController+FeaturedImageUpload.swift

```diff
@@ -5,16 +5,17 @@ extension PostSettingsViewController {
 
     @objc func setFeaturedImage(asset: PHAsset) {
         let media = MediaCoordinator.shared.addMedia(from: asset, to: self.apost)
+        apost.featuredImage = media
         setupObservingOf(media: media)
     }
 
     @objc func setFeaturedImage(media: Media) {
-        if media.hasRemote {
-            apost.featuredImage = media
+        if !media.hasRemote {
+            MediaCoordinator.shared.retryMedia(media)
+            setupObservingOf(media: media)
             return
         }
-        MediaCoordinator.shared.retryMedia(media)
-        setupObservingOf(media: media)
+        apost.featuredImage = media
```

Comment:
Not sure if my previous comment from here showed up – do we need to move this line to the top of the method? Previously we were setting `featuredImage` both whether the media had a remote or not, and now it's only if there's no remote. Don't we always need to set it?

Reply:
You are correct! Thanks for spotting this!